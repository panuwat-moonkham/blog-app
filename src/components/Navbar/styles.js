import styled from 'styled-components'

export const Container = styled.div`
  width: 100%;
  position: fixed;
  z-index: 10000;
  background-color: ${props => (props.isScroll ? '#fff' : 'transparent')};
  color: ${props => (props.isScroll ? '#000' : '#fff')};
  transition: background-color 0.5s;
`
export const Wrapper = styled.div`
  display: flex;
  flex: 1;
  padding: 2rem;
`

export const Logo = styled.div``

export const ActionContainer = styled.ul`
  display: flex;
  flex: 1;
  justify-content: flex-end;
`

export const Action = styled.li`
  padding: 0 1rem;
`