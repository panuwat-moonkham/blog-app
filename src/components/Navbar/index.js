import React from 'react'
import { Link } from 'react-router-dom'
import { Container, Wrapper, Logo, ActionContainer, Action } from '../Navbar/styles'
import useScroll from '../../utils/useScroll'

function NavBar() {
  const isScroll = useScroll({ scrollRange: 20 })

  return (
    <Container isScroll={isScroll}>
      <Wrapper>
        <Logo>Logo</Logo>
        <ActionContainer>
          <Action>
            <Link to="/">Home</Link>
          </Action>
          <Action>
            <Link to="/about">About</Link>
          </Action>
          <Action>
            <Link to="/contact">Contact</Link>
          </Action>
          <Action>
            <Link to="/story">Story</Link>
          </Action>
        </ActionContainer>
      </Wrapper>
    </Container>
  )
}

export default NavBar
