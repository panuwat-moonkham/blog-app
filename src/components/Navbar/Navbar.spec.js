//global describe
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import Navbar from '.'

describe('Navbar', () => {
  test('should contains logo text', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(getByText(/logo/gi)).toBeInTheDocument()
  })

  test('should have link navugagate to home, about, contact and story', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(getByText(/home/gi)).toBeInTheDocument()
    expect(getByText(/home/gi).closest('a')).toHaveAttribute('href', '/')

    expect(getByText(/about/gi)).toBeInTheDocument()
    expect(getByText(/about/gi).closest('a')).toHaveAttribute('href', '/about')

    expect(getByText(/contact/gi)).toBeInTheDocument()
    expect(getByText(/contact/gi).closest('a')).toHaveAttribute('href','/contact')

    expect(getByText(/story/gi)).toBeInTheDocument()
    expect(getByText(/story/gi).closest('a')).toHaveAttribute('href', '/story')
  })

  test('should render layout correcyly', () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
