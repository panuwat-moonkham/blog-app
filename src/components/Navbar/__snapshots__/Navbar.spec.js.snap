// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render layout correcyly 1`] = `
<div>
  <div
    class="sc-bdnylx jEByXK"
  >
    <div
      class="sc-gtssRu gwgwje"
    >
      <div
        class="sc-dlnjPT"
      >
        Logo
      </div>
      <ul
        class="sc-hKFyIo ljvqsi"
      >
        <li
          class="sc-eCApGN XRNFo"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="sc-eCApGN XRNFo"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="sc-eCApGN XRNFo"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
        <li
          class="sc-eCApGN XRNFo"
        >
          <a
            href="/story"
          >
            Story
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
